include/pagmo/detail/task_queue.hpp

```diff
@@ -61,11 +61,14 @@ struct PAGMO_DLL_PUBLIC task_queue {
     }
 
     // Data members.
-    bool m_stop;
-    std::condition_variable m_cond;
+    bool m_stop, m_park;
+    std::condition_variable m_cond, m_parked;
     std::mutex m_mutex;
     std::queue<task_type> m_tasks;
     std::thread m_thread;
+
+    static void park(std::unique_ptr<task_queue> &&tq);
```

Comment:
Can you remove the name of the parameter (``tq``) since this is just the function declaration?

Also, if this function is used only from within the ``.cpp`` files of the pagmo library, you can mark this as ``PAGMO_DLL_LOCAL``.